syntax = "proto3";

package protousers;

message UserResponse {
    int32 id = 1;
    string username = 2;
    string phone = 3;
    string email = 4;
    string first_name = 5;
    string last_name = 6;
    string middle_name = 7;
    string activity = 8;
    string status = 9;
    bool email_confirmed = 10;
    bool phone_confirmed = 11;
    string last_seen = 12;
    string original_avatar_url = 13;
    string converted_avatar_url = 14;
}

message GetUserByIdRequest {
    int32 id = 1;
}

message GetUserByUsernameRequest {
    string username = 1;
}

message GetUserByEmailRequest {
    string email = 1;
}

message GetUserByTokenRequest {
    string token = 1;
}

service Users {
    rpc GetUserById(GetUserByIdRequest) returns (UserResponse) {}
    rpc GetUserByUsername(GetUserByUsernameRequest) returns (UserResponse) {}
    rpc GetUserByEmail(GetUserByEmailRequest) returns (UserResponse) {}
    rpc GetUserByToken(GetUserByTokenRequest) returns (UserResponse) {}
    rpc GetUserByRefreshToken(GetUserByTokenRequest) returns (UserResponse) {}
}
